### PR TITLE
Add group badges to list view

### PR DESCRIPTION
**Pull request by EXtremeExploit**
Whenever there are no badges the class takes some extra space, so i check for any groups whenever the user changes and show/hide according to if there are groups present
image

Still there is some extra spacing at the edges of the class but i am still learning how stuff works c:

### CODE REVIEW

**Comment by peppy** [comment]
The supporter badge feels a bit big in comparison, but maybe it's okay. @arflyte can you confirm intentions here?

**Comment by stanriders** [comment]
It should really be a part of https://github.com/ppy/osu/issues/22130. I haven't touched it yet because the designs turned out to be not quite polished, so I'm waiting for flyte to update it before going any further, including updating the list/brick views.

**Comment by peppy** [comment]
I see. @stanriders what's your thoughts on this PR then? Should we look to merge as-is then fix the design issues as a follow up?

**Comment by stanriders** [comment]
Well from the design side the supporter button definitely needs to be the same size as badges, it's something that should be done in this PR. Other than that - list view badges are supposed to be combined too, but since there's a lot of free space I think we can merge it as is for now and change it lated when flyte decides on how combining should work.

The code can benefit from some restructuring for future PRs tho

File: osu.Game/Users/ExtendedUserPanel.cs [hunk 1]
```diff
@@ -97,6 +98,13 @@ protected FillFlowContainer CreateStatusMessage(bool rightAlignedChildren)
             return statusContainer;
         }
 
+        protected FillFlowContainer CreateGroupBadges()
```
Comment (stanriders) [inline osu.Game/Users/ExtendedUserPanel.cs:101]: I don't think this is required? You can just add it directly in the `UserListPanel`

File: osu.Game/Overlays/Profile/Header/Components/GroupBadgeFlow.cs [hunk 2]
```diff
@@ -25,7 +25,12 @@ public GroupBadgeFlow()
                 Clear(true);
 
                 if (user.NewValue?.Groups != null)
+                {
                     AddRange(user.NewValue.Groups.Select(g => new GroupBadge(g)));
+                    Show();
+                }
+                else
+                    Hide();
```
Comment (stanriders) [inline osu.Game/Overlays/Profile/Header/Components/GroupBadgeFlow.cs:33]: This shouldn't really be here as the container will be 0x0 and invisible anyway
image

File: osu.Game/Users/ExtendedUserPanel.cs [hunk 3]
```diff
@@ -97,6 +98,13 @@ protected FillFlowContainer CreateStatusMessage(bool rightAlignedChildren)
...
+        {
+            var groupBadgeFlow = new GroupBadgeFlow();
+            groupBadgeFlow.User.Value = User;
+            return groupBadgeFlow;
```
Comment (stanriders) [inline osu.Game/Users/ExtendedUserPanel.cs:105]:
```suggestion
            return new GroupBadgeFlow
            {
                User = { Value = User }
            };
```

File: osu.Game/Users/UserListPanel.cs [hunk 4]
```diff
@@ -67,6 +67,11 @@ protected override Drawable CreateLayout()
                             {
                                 username.Anchor = Anchor.CentreLeft;
                                 username.Origin = Anchor.CentreLeft;
+                            }),
+                            CreateGroupBadges().With(badges =>
```
Comment (stanriders) [inline osu.Game/Users/UserListPanel.cs:71]: If you don't want fill flow container to add spacing when there are no badges then add the `GroupBadgeFlow` after the layout creation with `if (User.Groups?.Length > 0)`, similar to how supporter badge is added.

**Review by stanriders — CHANGES_REQUESTED** [review]

File: osu.Game.Tests/Visual/Online/TestSceneFriendDisplay.cs [hunk 5]
```diff
@@ -56,7 +56,13 @@ public void TestOnline()
                 IsOnline = true,
                 Statistics = new UserStatistics { GlobalRank = 1111 },
                 CountryCode = CountryCode.JP,
-                CoverUrl = "https://osu.ppy.sh/images/headers/profile-covers/c6.jpg"
+                CoverUrl = "https://osu.ppy.sh/images/headers/profile-covers/c6.jpg",
```
Comment (stanriders) [inline osu.Game.Tests/Visual/Online/TestSceneFriendDisplay.cs:59]: There's a dedicated test scene for user panels - `TestSceneUserPanel`

**Comment by stanriders** [comment]
image

The code is fine, but design-wise this needs some fixing:

- Supporter button is bigger than badges
- Nickname centering is incorrect (seems like it was centered to the hearts in the supported badge?)

image
New designs have all icons the same size so I think we should go the same way here

**Comment by frenzibyte** [comment]
The nickname is probably not centred due to the extra space on top of the text for proper line alignment, which should be turned off for single labels. see `SpriteText.UseFullGlyphHeight`.

**Comment by peppy** [comment]
@EXtremeExploit see above, are you able to apply the changes to this PR?

**Comment by EXtremeExploit** [comment]
i've been trying but all my attempts to make them look the same without breaking didn't work, i need some advice on how to do it, else it could be merged without the size change

**Comment by stanriders** [comment]
> i've been trying but all my attempts to make them look the same without breaking didn't work, i need some advice on how to do it, else it could be merged without the size change

Set `SupporterIcon`'s `Height` to 16, add `username.UseFullGlyphHeight = false;` to the `CreateUsername().With(username =>{})` block
image

**Review by stanriders — APPROVED** [review]
Looks fine to me

**Review by peppy — APPROVED** [review]

**Pull request merged**